ql/termstructures/yield/compositezeroyieldstructure.hpp

```diff
@@ -0,0 +1,157 @@
+/* -*- mode: c++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*
+Copyright (C) 2000, 2001, 2002, 2003 RiskMap srl
+Copyright (C) 2007, 2008 StatPro Italia srl
+Copyright (C) 2017 Francois Botha
+
+This file is part of QuantLib, a free-software/open-source library
+for financial quantitative analysts and developers - http://quantlib.org/
+
+QuantLib is free software: you can redistribute it and/or modify it
+under the terms of the QuantLib license.  You should have received a
+copy of the license along with this program; if not, please email
+<quantlib-dev@lists.sf.net>. The license is also available online at
+<http://quantlib.org/license.shtml>.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+FOR A PARTICULAR PURPOSE.  See the license for more details.
+*/
+
+/*! \file compositezeroyieldstructure.hpp
+\brief Composite zero term structure
+*/
+
+#ifndef quantlib_composite_zero_yield_structure
+#define quantlib_composite_zero_yield_structure
+
+
+#include <ql/termstructures/yield/zeroyieldstructure.hpp>
+
+namespace QuantLib {
+    template <class BinaryFunction>
+    class CompositeZeroYieldStructure : public ZeroYieldStructure {
+      public:
+          CompositeZeroYieldStructure(const Handle<YieldTermStructure>& h1,
+                                      const Handle<YieldTermStructure>& h2,
+                                      const BinaryFunction& f,
+                                      Compounding comp = Continuous,
+                                      Frequency freq = NoFrequency,
+                                      const DayCounter& dc = DayCounter());
+
+          //! \name YieldTermStructure interface
+          //@{
+          DayCounter dayCounter() const;
+          Calendar calendar() const;
+          Natural settlementDays() const;
+          const Date& referenceDate() const;
+          Date maxDate() const;
+          Time maxTime() const;
+          //@}
+          //! \name Observer interface
+          //@{
+          void update();
+          //@}
+      protected:
+        //! returns the spreaded zero yield rate
+        Rate zeroYieldImpl(Time) const;
+        //! returns the spreaded forward rate
+        /* This method must disappear should the spread become a curve */
+        Rate forwardImpl(Time) const;
```

Comment:
`forwardImpl` is not in the interface of the base class `ZeroYieldStructure`, so it's never used and it can be removed.

Reply:
I actually wondered about that. Why is it in `ZeroSpreadedTermStructure` then?

Reply:
Good question.  In fact, it can be removed there, too.  I think it's left over from an ancient version of the code ("ancient" as in "from more than 10 years ago").